scripts/tools/FileSerializerMigration.php

```diff
@@ -0,0 +1,195 @@
+            $this->migrationHelper->migrationInformation['old_resource_migration_count'],
+            $this->migrationHelper->migrationInformation['migrated_count']
+        )));
+
+        if (count($this->migrationHelper->failedResources)) {
+            $this->reportMigrationErrors();
+        }
+
+        if ($this->migrationHelper->updateFileSerializer()) {
+            $this->report->add(Report::createSuccess(
+                'Successfully updated FileReferenceSerializer service to use UrlFileSerializer service'
+            ));
+        }
+
+        $end = microtime(true);
+        $this->report->add(Report::createSuccess(
+            sprintf('Completed file migration process. \'Process took %s Seconds to complete\'', round($end - $start, 2))
+        ));
+
+        if (!$this->isWetRun()) {
+            $this->report->add(Report::createFailure('Use the --wet-run (-w) parameter to execute reference migration.'));
+        }
+
+        return $this->report;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function provideUsage()
+    {
+        return [
+            'prefix' => 'h',
+            'longPrefix' => 'help',
+            'description' => 'Prints this help dialog'
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function provideOptions()
+    {
+        return [
+            'wetRun' => [
+                'flag' => true,
+                'prefix' => 'w',
+                'longPrefix' => 'wet-run',
+                'description' => 'Add this flag to migrate the references, as opposed to just listing them (dry/wet run)'
+            ],
+            'resourceLimit' => [
+                'prefix' => 'l',
+                'longPrefix' => 'limit',
+                'cast' => 'integer',
+                'defaultValue' => self::RESOURCE_LIMIT,
+                'description' => 'How many items should be processed each run?'
+            ]
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function provideDescription()
+    {
+        return 'Migration script to migrate Resource file references to URL file references';
+    }
+
+    /**
+     * Check whether we're running a wet run or not.
+     *
+     * @return bool
+     */
+    private function isWetRun()
+    {
+        return $this->hasOption('wetRun');
+    }
+
+    /**
+     * @param int $resourceOffset
+     * @return void
+     * @throws InvalidServiceManagerException
+     * @throws \oat\generis\model\fileReference\FileSerializerException
+     * @throws \oat\search\base\exception\SearchGateWayExeption
+     * @throws common_exception_Error
+     */
+    private function migrate()
+    {
+        $oldResourcesData = $this->migrationHelper->getOldResourcesData();
+        $oldResourceCount = $this->migrationHelper->migrationInformation['old_resource_count'];
+
+        $chuncks = array_chunk($oldResourcesData, self::RESOURCE_LIMIT);
```

Comment:
the idea of chunking is not to load all resources into memory, not sure there is a benefit of chunking the data in this case as you will only iterate over the chunk in the helper